rcl/include/rcl/publisher.h

```diff
@@ -465,6 +465,33 @@ rcl_publisher_get_subscription_count(
   const rcl_publisher_t * publisher,
   size_t * subscription_count);
 
+/// Get the actual qos settings of the publisher.
+/**
+ * Used to get the actual qos settings of the publisher.
+ * This resolves what is the actual value of RMW_*_SYSTEM_DEFAULT.
```

Comment:
Can you make this more thorough, touching on the fact that these system defaults can only be resolved after creating the publisher and that they may be set to unknown if the underlying default setting cannot be represented in ROS terms.